File: frontend/src/types/resume.ts
export type ParseStatus =
  | 'NOT_PARSED'
  | 'PENDING'
  | 'PROCESSING'
  | 'COMPLETED'
  | 'FAILED'

export interface Resume {
  id: string
  label: string
  versionNo: number
  fileName: string
  contentType: string | null
  fileSizeBytes: number | null
  storageUrl: string
  primary: boolean
  parseStatus: ParseStatus
  parsedAt: string | null
  parseError: string | null
  notes: string | null
  createdAt: string
  updatedAt: string
  version: number
}

export interface CreateResumeRequest {
  label: string
  storageUrl: string
  fileName?: string | null
  contentType?: string | null
  fileSizeBytes?: number | null
  notes?: string | null
  primary?: boolean
}

export interface UpdateResumeRequest {
  label?: string
  notes?: string | null
  primary?: boolean
}

File: backend/resume-service/src/main/java/com/careerflow/resumeservice/resume/model/Resume.java
package com.careerflow.resumeservice.resume.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import jakarta.persistence.Version;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "resumes")
@Getter
@Setter
@NoArgsConstructor
public class Resume {

    @Id
    private UUID id;

    @Column(name = "user_id", nullable = false)
    private String userId;

    @Column(name = "label", nullable = false)
    private String label;

    @Column(name = "version_no", nullable = false)
    private int versionNo;

    @Column(name = "file_name", nullable = false)
    private String fileName;

    @Column(name = "content_type")
    private String contentType;

    @Column(name = "file_size_bytes")
    private Long fileSizeBytes;

    @Column(name = "storage_url", nullable = false)
    private String storageUrl;

    @Column(name = "storage_key", nullable = false)
    private String storageKey;

    @Column(name = "is_primary", nullable = false)
    private boolean primary;

    @Enumerated(EnumType.STRING)
    @Column(name = "parse_status", nullable = false)
    private ParseStatus parseStatus;

    @Column(name = "parsed_at")
    private Instant parsedAt;

    @Column(name = "parse_error")
    private String parseError;

    @Column(name = "notes")
    private String notes;

    @Column(name = "created_at", nullable = false)
    private Instant createdAt;

    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;

    @Version
    @Column(name = "version", nullable = false)
    private Long version;

    @PrePersist
    void onCreate() {
        if (id == null) {
            id = UUID.randomUUID();
        }
        Instant now = Instant.now();
        createdAt = now;
        updatedAt = now;
        if (parseStatus == null) {
            parseStatus = ParseStatus.NOT_PARSED;
        }
    }

    @PreUpdate
    void onUpdate() {
        updatedAt = Instant.now();
    }
}
